logic/peers.js

```diff
@@ -108,6 +110,42 @@ Peers.prototype.get = function(peer) {
 	return self.peersManager.getByAddress(peer.string);
 };
 
+/**
+ * Ban peer by setting BANNED state. Make it temporarily using banManager.
+ *
+ * @param {Peer} peer
+ *
+ */
+Peers.prototype.ban = function(peer) {
+	peer.state = Peer.STATE.BANNED;
+	// Banning peer can only fail with ON_MASTER.UPDATE.INVALID_PEER error.
+	// Happens when we cannot obtain the proper address of a given peer.
+	// In such a case peer will be removed.
+	if (self.upsert(peer) === false) {
+		self.remove(peer);
+		library.logger.info('Attempt to ban a peer failed and resulted with removal');
```

Comment:
That should be logged before `self.remove(peer)`, because that function is logging also.